app/controllers/application_controller/queries.rb

```diff
@@ -335,9 +384,9 @@ def redirect_to_next_object(method, model, id)
     query =  next_params[:query]
 
     # Redirect to the show_object page appropriate for the new object.
-    redirect_to(add_query_param({ controller: object.show_controller,
-                                  action: object.show_action,
-                                  id: id }, query))
+    redirect_to({ controller: object.show_controller,
+                  action: object.show_action,
+                  id:, q: get_query_param(query) })
```

Comment:
Is there a plan to rename this method?  I'm not a fan of method names that start with "get_" and I believe avoiding them is also standard Rails practice.  Is there a compelling reason it's not just called `q_param`?  It could even be off the Query class which might be a better place than ApplicationController::Queries.  Doesn't need to be this PR, but seems like a reasonable cleanup to do at some point.